test/e2e/rc.go

```diff
@@ -26,17 +26,41 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
-	"github.com/golang/glog"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
+var _ = Describe("ReplicationController", func() {
+	var (
+		c *client.Client
+	)
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should serve a basic image on each replica with a public image", func() {
+		ServeImageOrFail(c, "basic", "kubernetes/serve_hostname:1.1")
+	})
+
+	It("should serve a basic image on each replica with a private image", func() {
+		if testContext.provider != "gce" && testContext.provider != "gke" && testContext.provider != "aws" {
```

Comment:
This is starting to get a bit long & ugly. How about a switch instead?

switch testContext.provider {
  case "gce", "gke", "aws":
    ServeImageOfFail(...)
  default:
    Logf("Skipping...")
}